llvm-17.yaml

```diff
@@ -0,0 +1,819 @@
+          done
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/lib/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/lib/libfindAllSymbols.a ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/cmake
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/lib/cmake/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/cmake/
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/share
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/share/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/share/
+
+          mkdir -p ${{targets.contextdir}}/${{vars.llvm-prefix}}/include
+          mv ${{targets.destdir}}/${{vars.llvm-prefix}}/include/*clang* ${{targets.contextdir}}/${{vars.llvm-prefix}}/include/
+
+          mkdir -p ${{targets.contextdir}}/usr/lib/clang
+          ln -s ../llvm-${{vars.major-version}}/lib/clang/${{vars.major-version}} ${{targets.contextdir}}/usr/lib/clang/${{vars.major-version}}
+
+          # Make libclang public
+          soname="libclang.so.${{vars.major-version}}"
+          sonamefull="libclang.so.${{package.version}}"
+
+          # Move from private Clang lib to public lib
+          mkdir -p ${{targets.contextdir}}/usr/lib/
+          mv ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/$sonamefull ${{targets.contextdir}}/usr/lib/
+
+          # Correct symlinks to libclang
+          ln -s $sonamefull ${{targets.contextdir}}/usr/lib/$soname
+          ln -sf ../../$sonamefull ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/$soname
+          ln -sf ../../$sonamefull ${{targets.contextdir}}/${{vars.llvm-prefix}}/lib/$sonamefull
+    test:
+      pipeline:
+        - uses: test/tw/ldd-check
+        - uses: test/compiler-hardening-check
+          with:
+            cc: clang-${{vars.major-version}}
+        - runs: |
+            amdgpu-arch --version
+            amdgpu-arch --help
+            clang --version
+            clang --help
+            clang++ --version
+            clang++ --help
+            clang-${{vars.major-version}} --version
+            clang-${{vars.major-version}} --help
+            clang-cl --version
+            clang-cpp --version
+            clang-cpp --help
+            clang-include-cleaner --version
+            clang-include-cleaner --help
+            clang-linker-wrapper --version
+            clang-linker-wrapper --help
+            clang-pseudo --version
+            clang-pseudo --help
+            nvptx-arch --version
+            nvptx-arch --help
+
+  - name: lld-${{vars.major-version}}
+    description: LLVM linker
+    dependencies:
+      provides:
+        - llvm-lld-${{vars.major-version}}=${{package.full-version}}
+      runtime:
+        - libLLVM-17=${{package.full-version}}
```

Comment:
*nit* in llvm-16 we use ${{vars.major-version}} - feels like we should do the same here